frontend/src/app/components/channel/channel.controller.ts

```diff
@@ -6,53 +6,118 @@ export interface IChannelScope extends ng.IScope {
   users: IUser[];
   name: string;
   selfElement: ng.IAugmentedJQuery;
+  programElement: HTMLVideoElement;
+  previewElement: HTMLVideoElement;
 }
 
 /** @ngInject */
 export class ChannelController {
-  users: IUser[];
-  previewUserLogin: string;
-  scope: IChannelScope;
-  toastr: any;
   channel: ChannelService;
-  janus: JanusVideoRoomService;
+
   name: string;
+  users: IUser[];
+  hotkey: string;
+
+  usersByLogin: { [login: string]: IUser } = {};
+  onlineUsers: IUser[] = [];
+  previewUser: IUser = null;
+  programUser: IUser = null;
+
+  constructor(public $scope: IChannelScope,
+              public $timeout: ng.ITimeoutService,
+              public $document: any,
+              public janus: JanusVideoRoomService) {
+
+    // Mapping users by login for conveniency
+    this.users.forEach((user: IUser) => {
+      this.usersByLogin[user.login] = user;
+    });
 
-  constructor($scope: IChannelScope,
-              janus: JanusVideoRoomService,
-              toastr: any, config: any) {
-    this.scope = $scope;
-    this.janus = janus;
+    this.bindHotkey();
 
     this.janus.registerChannel({
       name: this.name,
-      users: this.users.map((u) => { return u.login; }),
-      joinedCallback: (login: string) => { this.userJoined(login); },
-      leftCallback: (login: string) => { this.userLeft(login); }
+      users: this.users.map((user: IUser) => { return user.login; }),
+      // Wrapping into $timeout for syncing the UI
+      joinedCallback: (login: string) => {
+        $timeout(() => {
+          this.userJoined(login);
+        });
+      },
+      leftCallback: (login: string) => {
+        $timeout(() => {
+          this.userLeft(login);
+        });
+      }
     });
   }
 
+  bindHotkey() {
+    if (this.hotkey) {
+      this.$document.bind('keydown', (e: KeyboardEvent) => {
+        if (e.keyCode === this.hotkey.charCodeAt(0)) {
+          this.$timeout(() => {
+            this.next();
+          });
+        }
+      });
+    }
+  }
+
   userJoined(login: string) {
-    // TODO: Moment actually better to register while user logges in
-    // from user point of view, not from shidur point of view
-    // this.users[login].joined = moment();
-
-    // Means he sends video/audio to janus
-    // Now we decide to get his video/audio here or not.
-    // If program or preview ==> get stream + show on video element
-    if (!this.previewUserLogin) {
-      this.previewUserLogin = login;
-      var element = this.scope.selfElement.find('.preview').get(0);
-      this.janus.attachRemoteHandle(login, element);
+    // TODO: The timestamp should be better taken from Janus point of view
```

Comment:
Don't know a straightforward way to do it for now, also current approach works in most cases.